File: C/airportTester.c
/**
 * Author Max Schessler
 * Date 2024-11-15
 * 
 * This file holds my Airport Lib testing program.
 */

#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include <math.h>
#include "airport.h"

int main() {

    // testing createAirport
    Airport *a1 = createAirport("KATL", "large_airport", "Hartsfield-Jackson Atlanta International Airport", 33.6367, -84.4279, 1026, "Atlanta", "US");
    Airport *a2 = createAirport("KJFK", "large_airport", "John F Kennedy International Airport", 40.6398, -73.7789, 13, "New York", "US");
    Airport *a3 = createAirport("KORD", "large_airport", "Chicago O'Hare International Airport", 41.9786, -87.9048, 672, "Chicago", "US");

    freeAirport(a1);
    freeAirport(a2);
    freeAirport(a3);

    // testing initAirport
    a1 = (Airport *) malloc(sizeof(Airport));
    a2 = (Airport *) malloc(sizeof(Airport));
    a3 = (Airport *) malloc(sizeof(Airport));
    initAirport(a1, "KATL", "large_airport", "Hartsfield-Jackson Atlanta International Airport", 33.6367, -84.4279, 1026, "Atlanta", "US");
    initAirport(a2, "KJFK", "large_airport", "John F Kennedy International Airport", 40.6398, -73.7789, 13, "New York", "US");
    initAirport(a3, "KORD", "large_airport", "Chicago O'Hare International Airport", 41.9786, -87.9048, 672, "Chicago", "US");


    // testing airportToString
    char *a1Str = airportToString(a1);
    char *a2Str = airportToString(a2);
    char *a3Str = airportToString(a3);
    printf("%s\n", a1Str);
    printf("%s\n", a2Str);
    printf("%s\n", a3Str);

    free(a1Str);
    free(a2Str);
    free(a3Str);

    // testing getAirDistance
    double dist1 = getAirDistance(a1, a2);
    double dist2 = getAirDistance(a2, a3);
    double dist3 = getAirDistance(a1, a3);
    printf("Distance between KATL and KJFK: %f\n", dist1);
    printf("Distance between KJFK and KORD: %f\n", dist2);
    printf("Distance between KATL and KORD: %f\n", dist3);

    Airport *stops = (Airport *) malloc (sizeof(Airport) * 3);
    stops[0] = *a1;
    stops[1] = *a2;
    stops[2] = *a3;

    double estimate = getEstimatedTravelTime(stops, 3, 300, 1);
    printf("Estimate: getEstimatedTravelTime(stops, 3, 300, 1): %.2f\n", estimate);
    estimate = getEstimatedTravelTime(stops, 3, 300, 3);
    printf("Estimate: getEstimatedTravelTime(stops, 3, 300, 3): %.2f\n", estimate);
    estimate = getEstimatedTravelTime(stops, 3, 600, 1);
    printf("Estimate: getEstimatedTravelTime(stops, 3, 600, 1): %.2f\n", estimate);

    freeAirport(a1);
    freeAirport(a2);
    freeAirport(a3);
    free(stops);

    return 0;
}

File: java/src/main/java/unl/soc/AirportTester.java
/**
 * Author Max Schessler
 * Date 2024-11-12
 */


package unl.soc;

import java.util.Arrays;
import java.util.List;

public class AirportTester {
    public static void main(String[] args) {

        //Testing constructors - will use these objects for more testing
        Airport a1 = new Airport("KATL", "large_airport", "Hartsfield-Jackson Atlanta International Airport",
        33.6367, -84.4279, 1026, "Atlanta", "US");

        Airport a2 = new Airport("KJFK", "large_airport", "John F Kennedy International Airport",
                40.6398, -73.7789, 13, "New York", "US");

        Airport a3 = new Airport("KORD", "large_airport", "Chicago O'Hare International Airport",
                41.9786, -87.9048, 672, "Chicago", "US");

        List<Airport> stops = Arrays.asList(a1, a2, a3);

        //Testing toString method
        System.out.println("\nMy Airports: ");
        System.out.println(a1.toString());
        System.out.println(a2.toString());
        System.out.println(a3.toString());

        //Testing getAirDistance method
        System.out.println("\nAir Distances: ");
        System.out.printf("Air Distance from %s to %s: %f\n", a1.name, a2.name, a1.getAirDistance(a2));
        System.out.printf("Air Distance from %s to %s: %f\n", a1.name, a3.name, a1.getAirDistance(a3));
        System.out.printf("Air Distance from %s to %s: %f\n", a2.name, a3.name, a2.getAirDistance(a3));

        //Testing getTravelTime method
        System.out.println("\nTravel Times: ");
        double travelTime = Airport.getEstimatedTravelTime(stops, 500, 1);
        System.out.printf("Estimated Travel Time for flight with 500 kmp and 3 stops of 1 hour: %.2f hours\n", travelTime);
        travelTime = Airport.getEstimatedTravelTime(stops, 500, 2);
        System.out.printf("Estimated Travel Time for flight with 500 kmp and 3 stops of 2 hour: %.2f hours\n", travelTime);
        travelTime = Airport.getEstimatedTravelTime(stops, 1000, 1);
        System.out.printf("Estimated Travel Time for flight with 1000 kmp and 3 stops of 1 hour: %.2f hours\n", travelTime);

    }
}